tests/dispatch/tests.py

```diff
@@ -53,22 +55,43 @@ def assertTestIsClean(self, signal):
         self.assertFalse(signal.has_listeners())
         self.assertEqual(signal.receivers, [])
 
-    def test_exact(self):
+    @override_settings(DEBUG=True)
+    def test_cannot_connect_no_kwargs(self):
+        def receiver_no_kwargs(sender):
+            pass
+        with self.assertRaisesRegex(ValueError, r'must accept keyword arguments'):
```

Comment:
Yup, just needed to make it a `SimpleTestCase` rather than `unittest.TestCase` (at YPlan we use https://github.com/adamchainz/pytest-restrict to disallow `unittest.TestCase` and only have tests in our local versions of Django's TestCase classes - could be useful option for Django's test suite?)